### DISCUSSION TITLE
Provide search functionality

### DISCUSSION THREAD

**Discussion by sampittko** [payloadcms/payload#162] [Feature Requests & Ideas]
# Missing search functionality

Documentation is pretty robust and there is no way to browse quickly through it.

## Additional Details

`cmd + K` shortcut should display dialog with search bar and results that link to the destination with discovered information (e.g. [TailwindCSS](https://tailwindcss.com)).

**Comment by jmikrut** [comment]
Hey @sampittko — you will be happy to hear that we are actually almost done implementing Algolia Docsearch which is exactly what Tailwind uses!

Going to move this to a Discussion until we launch the search. Until then, keep an eye out!

**Comment by Ontopic** [comment]
Better search on docs is much appreciated. As sampittko said; they're pretty robust already.

Problem for me is / was mostly that the good stuff is "hidden" away in the demo folder of the repo or answers to issues here or there on Github. And of course James' videos - which I loved - but it's probably the first time I actually sat through a Youtube video about coding (FOMO, since no docs); usually I prefer to read the actual code and see examples. Extending the docs with samples and more real life demo's would I think shed a lot more light on how to approach certain issues and make it more approachable for starting devs. I think NextJS and Gatsby are both prime examples of this; lots of boilerplates and demos.

Right now I feel like I'm the only with Richard (he'll probably read this message 🙃) pursuing getting to the bottom of what's possible with Payload - which I'm thinking  is a lot - openly apart from the core team. I keep a list of open source projects (mostly extended starter repos) I found that use Payload, but it's kinda hard to find real life examples. Googling or searching for PayloadCMS on Github is also kinda hard, apparently kind a lot of repos contain the words payload and cms 😅

TLDR; I think we could on-board a lot of people with more demos. I'm personally eager to share thoughts and code with anyone interested, perhaps to be published as generalised example repos

Some examples of "hidden gems"
[https://github.com/payloadcms/payload/discussions/139](url) 
[https://github.com/payloadcms/payload/discussions/93](url)

**Reply by DanRibbens** [reply]
I'm adding `?locale=all` to the docs right now.
I have a draft of a blog post talking about #93 customizing field elements UI. The plan is to keep making blog posts that help with knowledge transfer that will become a full section of guides.

We also discussed making a **awesome-payload** repo to curate a list of useful resources. Your feedback reinforces the value in doing that.

Now, back to that search functionality! :)

**Reply by Ontopic** [reply]
_awesome-payload ftw and shutting up now_

**Comment by Ontopic** [comment]
Also just phishing for more hidden gems 😈🤑

**Comment by jmikrut** [comment]
Search is now live! It's a first implementation, so it'll get better from here, but take a look! What do you think? @sampittko
